### PR TITLE
Fix BigQuery#listDatasets to include dataset location in the response

### LINKED ISSUE #3237: BigQuery::listDatasets API response does not populate Dataset::location field
#### Steps to reproduce

  1. Invoke BigQuery::listDatasets
  2. Invoking Dataset::getLocation on the response shall return null

#### Code example

```java
String projectId = "bigquery-public-data";
FileInputStream stream = new FileInputStream(<creds json path>);

BigQuery bigquery =
    BigQueryOptions.newBuilder()
        .setCredentials(ServiceAccountCredentials.fromStream(stream))
        .build()
        .getService();

Page<Dataset> page = bigquery.listDatasets(projectId);
for (Dataset dataset : page.iterateAll()) {
  assertThat(dataset.getLocation()).isNull();
}
```

#### Any additional information below
- The field is available in the `com.google.api.services.bigquery.model.Dataset#location` however, it is not mapped to `com.google.cloud.bigquery.Dataset#location by `HttpBigQueryRpc#LIST_TO_DATASET`

https://github.com/googleapis/google-api-java-client-services/blob/9ab9bf5d1cb81e24c33dfd824fd5645305e0d108/clients/google-api-services-bigquery/v2/2.0.0/com/google/api/services/bigquery/model/Dataset.java#L211-L217

https://github.com/googleapis/java-bigquery/blob/43409c9fa879de27087763e2a2b04c4d0582429b/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/spi/v2/HttpBigQueryRpc.java#L88-L100

Thanks!

### PR DESCRIPTION
**Pull request by sumeetgajjar**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #3237 ☕️

### CODE REVIEW

**Comment by sumeetgajjar** [comment]
Hi @mrfaizal @PhongChuong - can you please review this PR and trigger cloud CI if the change looks satisfactory?

Thanks in advance.

**Comment by sumeetgajjar** [comment]
Hi @shollyman @PhongChuong @mrfaizal - A gentle ping to review this PR.
Thanks in advance.

File: google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java [hunk 1]
```diff
@@ -774,6 +774,15 @@ public class ITBigQueryTest {
...
       ImmutableSet.of("github_repos", "hacker_news", "noaa_gsod", "samples", "usa_names");
+  private static final Map<String, String> PUBLIC_NON_US_DATASETS =
+      ImmutableMap.<String, String>builder()
+          // Dataset url:
+          // https://console.cloud.google.com/bigquery?project=bigquery-public-data&ws=!1m4!1m3!3m2!1sbigquery-public-data!2sgnomAD_asiane1
+          .put("gnomAD_asiane1", "asia-northeast1")
+          // Dataset url:
+          // https://console.cloud.google.com/bigquery?project=bigquery-public-data&ws=!1m4!1m3!3m2!1sbigquery-public-data!2sgnomAD_eu
+          .put("gnomAD_eu", "EU")
+          .build();
```
Comment (PhongChuong) [inline google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:785]: It'll make the validation earlier if you combine PUBLIC_DATASETS into PUBLIC_NON_US_DATASETS.
I.e.,
private static final Map<String, String> PUBLIC_DATASETS_LOCATION =
...
.put("gnomAD_asiane1", "asia-northeast1")
.put("gnomAD_eu", "EU")
.put("github_repos", "US")
...
.build();

Then for validation:
for (String type : PUBLIC_DATASETS.keySet()) {
assertTrue(datasetNames.contains(type));
assertEquals(PUBLIC_DATASETS_LOCATION.get(type), datasetLocation.get(type));
}

**Review by PhongChuong — COMMENTED** [review]
Thank you for contributing.
Just a minor suggestion to the integration test.

File: google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java (hunk 1 above)
Comment (sumeetgajjar) [reply google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java]: Thanks for the suggestion @PhongChuong - addressed it in the latest commit.

**Review by PhongChuong — APPROVED** [review]

**Comment by PhongChuong** [comment]
Approved. Thanks for contributing.
Sorry for the delay.
Running the integration test now and it should be good to go.

**Pull request merged**